### PR TITLE
[14.0][FIX] upgrade_analysis: merged modules affecting noupdate files

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Due to https://github.com/OCA/server-tools/pull/2057, noupdate data now considers merged modules. But the fix was incomplete, as the diff files between merged modules and local_module were rewritting already generated diff between original remote_module and local_module.

### CODE REVIEW

**Comment by StefanRijnhart** [comment]
Will you also revert the functionality of `get_xml_records` to return xml data instead of paths? Otherwise we will have a problem with OpenUpgrade 15.

**Comment by MiquelRForgeFlow** [comment]
> Will you also revert the functionality of get_xml_records to return xml data instead of paths?

I could do that if there was a problem. As the issue of the noupdate data don't affect if we do the 'paths' approach or 'files' approach, for now I decided to maintain the 'paths' approach (in order to include the files that have a different encoding).

> Otherwise we will have a problem with OpenUpgrade 15

Which problem?

**Comment by StefanRijnhart** [comment]
To perform the noupdate data comparison in upgrade_analysis in Odoo 15, it needs this module in version 14 to provide the XML data (the remote paths to the XML data are useless). It's the same discussion as https://github.com/OCA/OpenUpgrade/pull/2481#discussion_r616535367

**Comment by pedrobaeza** [comment]
I also think that providing whole XML data instead of paths is better for isolating each environment (previous version and this one). Think for example with Doodba, that each Odoo is isolated with no access to their local file paths.

**Comment by MiquelRForgeFlow** [comment]
Yeah, but the upgrade_analysis (of v15) will be created doing a forward port of upgrade_analysis (of v14), which will have the same code (despite number version), and thus it will also have the same methods and approach.

**Comment by StefanRijnhart** [comment]
Oh now I see why you think it is not a problem, you are running the two versions side by side on the same machine on the same user account! Not everybody does that.

**Comment by MiquelRForgeFlow** [comment]
> Think for example with Doodba

Okey, that would be a reason, but I don't imagine anybody executing this analysis with doodba haha

**Comment by MiquelRForgeFlow** [comment]
> you are running the two versions side by side on the same machine on the same user account!

Of course. That's full control!

**Comment by StefanRijnhart** [comment]
Please revert to the version that is generic and not depending on your local setup.

**Comment by pedrobaeza** [comment]
@MiquelRForgeFlow what I don't imagine is doing this manually in your local machine without Doodba! You have to comply with both versions requirements, do the setup, etc. It's counter-productive.

As the data transmission way serves for both setups, please go with that standard.

**Comment by MiquelRForgeFlow** [comment]
Ok, reverted everywhere, all it's ready to merge.

**Comment by MiquelRForgeFlow** [comment]
@StefanRijnhart could you check? sure @pedrobaeza is waiting you hehe

File: upgrade_analysis/models/upgrade_record.py [hunk 1]
```diff
@@ -159,13 +159,17 @@ def get_xml_records(self, module):
...
             for xml_file in manifest[key]:
                 if not xml_file.lower().endswith(".xml"):
                     continue
                 parts = xml_file.split("/")
-                paths.append(os.path.join(addon_dir, *parts))
-        return paths
+                try:
+                    with open(os.path.join(addon_dir, *parts), "r") as xml_handle:
+                        files.append(xml_handle.read())
+                except UnicodeDecodeError:
```
Comment (StefanRijnhart) [inline upgrade_analysis/models/upgrade_record.py:176]: I'm thinking a warning would be good at this point

**Review by StefanRijnhart — APPROVED** [review]
~~Just a small non blocking comment.~~  Thanks for the follow-up!

File: upgrade_analysis/models/upgrade_record.py (hunk 1 above)
Comment (MiquelRForgeFlow) [reply upgrade_analysis/models/upgrade_record.py:176]: Done! :)

**Pull request merged**